controller/src/main/java/io/pravega/controller/server/eventProcessor/requesthandlers/CommitRequestHandler.java

```diff
@@ -180,10 +189,10 @@ public CommitRequestHandler(final StreamMetadataStore streamMetadataStore,
                                                         activeEpochRecord.getReferenceEpoch() == txnEpochRecord.getReferenceEpoch()) {
                                                     // If active epoch's reference is same as transaction epoch,
                                                     // we can commit transactions immediately
-                                                    return commitTransactions(scope, stream, new ArrayList<>(activeEpochRecord.getSegmentIds()), txnList, context)
+                                                    return commitTransactions(scope, stream, new ArrayList<>(activeEpochRecord.getSegmentIds()), txnList, context, timer)
```

Comment:
we dont need to pass timer to all methods, esp when its only exercised at one place after the last step.
 we should instead timer.getElapsed at the end of this method itself.

Reply:
I see two problem with that approach:
1) Granularity of transaction latency report: If we just get the time elapsed after this method, we are not measuring the time to commit individual transactions, but the whole batch. In the PR, we execute a `timer.getElapsed()` for each transaction, which looks more precise to me.
2) Completeness: I realized that the `rollTransactions()` method also invokes internally `commitTransactions()` (within `copyTxnEpochSegmentsAndCommitTxns()`). This means that we also need to cover this path to report all the transaction commit latency, otherwise we would be missing some transactions.